File: OMGStarWars/app/src/main/java/dev/radley/omgstarwars/models/Species.java
package dev.radley.omgstarwars.models;


import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.ArrayList;

/**
 * Species model represents a type of person or character within the Star Wars Universe.
 */
public class Species extends SWModel implements Serializable {

    public String classification;
    public String designation;
    public String language;

    @SerializedName("average_height")
    public String averageHeight;

    @SerializedName("average_lifespan")
    public String averageLifespan;

    @SerializedName("eye_colors")
    public String eyeColors;

    @SerializedName("hair_colors")
    public String hairColors;

    @SerializedName("skin_colors")
    public String skinColors;

    @SerializedName("homeworld")
    public String homeWorld;

    @SerializedName("people")
    public ArrayList<String> peopleUrls;

    @SerializedName("films")
    public ArrayList<String> filmsUrls;

    @Override
    public ArrayList<String> getRelatedFilms() {
        return filmsUrls;
    }
}

File: OMGStarWars/app/src/main/java/dev/radley/omgstarwars/models/Species.kt
package dev.radley.omgstarwars.models


import com.google.gson.annotations.SerializedName

import java.io.Serializable
import java.util.ArrayList

/**
 * Species model represents a type of person or character within the Star Wars Universe.
 */
class Species : SWModel(), Serializable {

    var classification: String = ""
    var designation: String = ""
    var language: String = ""

    @SerializedName("average_height")
    var averageHeight: String = ""

    @SerializedName("average_lifespan")
    var averageLifespan: String = ""

    @SerializedName("eye_colors")
    var eyeColors: String = ""

    @SerializedName("hair_colors")
    var hairColors: String = ""

    @SerializedName("skin_colors")
    var skinColors: String = ""

    @SerializedName("homeworld")
    var homeWorld: String = ""

    @SerializedName("people")
    var peopleUrls: ArrayList<String>? = null

    @SerializedName("films")
    override var relatedFilms: ArrayList<String>? = null
}